src/main/scala/Util.scala

```diff
@@ -84,25 +94,28 @@ object WideCounterModule {
 class WordSync[T <: Data](gen: T, lat: Int = 2) extends Module {
   val size = gen.getWidth
   val io = IO(new Bundle {
-    val in = Flipped(gen.chiselCloneType)
-    val out = gen.chiselCloneType
+    val in = Flipped(chiselTypeOf(gen))
+    val out = chiselTypeOf(gen)
     val tx_clock = Input(Clock())
   })
   val bin2gray = Module(new BinToGray(gen,io.tx_clock))
   bin2gray.io.bin := io.in
   val out_gray = ShiftRegister(bin2gray.io.gray, lat)
-  io.out := gen.cloneType.fromBits(
+  io.out := (
     (0 until size)
-      .map(out_gray.asUInt >> _.U)
-      .reduceLeft((a: UInt, b: UInt) => a^b))
+      .map{ case og => (out_gray.asUInt >> og.U).asUInt }
+      .reduceLeft((a: UInt, b: UInt) => (a^b).asUInt)
+    )
```

Comment:
This looks fine, but I'm only inspecting. I haven't tested any of this...

Reply:
I think this is beyond the scope of this PR, but this entire class looks broken to me. Gray coding is only safe for clock crossings if you're incrementing by 1, which does not appear to be what this is doing...